scripts/ldap/base_configure_ldap.py

```diff
@@ -0,0 +1,209 @@
+
+
+l.sasl_non_interactive_bind_s('EXTERNAL')
+dn="olcDatabase={0}config,cn=config"
+mod_attrs = [( ldap.MOD_REPLACE, 'olcRootDN', [b'cn=admin,cn=config'] )]
+try:
+    l.modify_s(dn,mod_attrs)
+except:
+    print('Error while modifying olcRootDN attribute for olcDatabase={0}config')
+
+mod_attrs = [( ldap.MOD_REPLACE, 'olcRootPW', pwd.encode("utf-8") )]
+try:
+    l.modify_s(dn,mod_attrs)
+except:
+    print('Error while modifying olcRootPW attribute for olcDatabase={0}config')
+
+mod_attrs = [( ldap.MOD_REPLACE, 'olcAccess', [b'{0}to * by dn.base="gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth" write by self write by * read'] )]
+try:
+    l.modify_s(dn,mod_attrs)
+except:
+    print('Error while modifying olcAccess attribute for olcDatabase={0}config')
+
+dn="olcDatabase={2}mdb,cn=config"
+mod_attrs = [( ldap.MOD_REPLACE, 'olcSuffix', [b'dc=seagate,dc=com'] )]
+try:
+    l.modify_s(dn,mod_attrs)
+except:
+    print('Error while modifying olcSuffix attribute for olcDatabase={2}mdb')
+
+mod_attrs = [( ldap.MOD_REPLACE, 'olcRootDN', [b'cn=admin,dc=seagate,dc=com'] )]
+try:
+    l.modify_s(dn,mod_attrs)
+except:
+    print('Error while modifying olcRootDN attribute for olcDatabase={2}mdb')
+
+mod_attrs = [( ldap.MOD_ADD, 'olcDbMaxSize', [b'10737418240'] )]
+try:
+    l.modify_s(dn,mod_attrs)
+except:
+    print('Error while modifying olcDbMaxSize attribute for olcDatabase={2}mdb')
+
+mod_attrs = [( ldap.MOD_REPLACE, 'olcRootPW', pwd.encode("utf-8") )]
+try:
+    l.modify_s(dn,mod_attrs)
+except:
+    print('Error while modifying olcRootPW attribute for olcDatabase={2}mdb')
+
+mod_attrs = [( ldap.MOD_REPLACE, 'olcAccess', [b'{0}to attrs=userPassword by self write by dn.base="cn=admin,dc=seagate,dc=com" write by anonymous auth by * none'] )]
+try:
+    l.modify_s(dn,mod_attrs)
+except:
+    print('Error while modifying olcAccess attribute for attrs=userPassword and dn.base=cn=admin,dc=seagate,dc=com in olcDatabase={2}mdb')
+
+mod_attrs = [( ldap.MOD_ADD, 'olcAccess', [b'{1}to * by dn.base="cn=admin,dc=seagate,dc=com" write by self write by * none'] )]
+try:
+    l.modify_s(dn,mod_attrs)
+except:
+    print('Error while modifying olcAccess attribute for * and dn.base=cn=admin,dc=seagate,dc=com in olcDatabase={2}mdb')
+
+mod_attrs = [( ldap.MOD_REPLACE, 'olcAccess', [b'{0}to attrs=userPassword by self write by dn.base="cn=sgiamadmin,dc=seagate,dc=com" write by anonymous auth by * none'] )]
+try:
+    l.modify_s(dn,mod_attrs)
+except:
+    print('Error while modifying olcAccess attribute for attrs=userPassword and dn.base=cn=sgiamadmin,dc=seagate,dc=com in olcDatabase={2}mdb')
+
+mod_attrs = [( ldap.MOD_ADD, 'olcAccess', [b'{1}to * by dn.base="cn=sgiamadmin,dc=seagate,dc=com" write by self write by * none'] )]
+try:
+    l.modify_s(dn,mod_attrs)
+except:
+    print('Error while modifying olcAccess attribute for * and dn.base=cn=sgiamadmin,dc=seagate,dc=com in olcDatabase={2}mdb')
+
+l.unbind_s()
+
+os.system('systemctl restart slapd')
```

Comment:
I think @SaumitraK531 recently removed some slapd restarts. Now we have only 1 restart after initial cleanup. Can you check if that can be addressed here.

Reply:
Sure. Will check and see if it can be removed

Reply:
fixed